src/pyhf/pdf.py

```diff
@@ -619,13 +630,21 @@ def expected_data(self, pars, return_by_sample=False):
         allfac = tensorlib.concatenate(factors + [nom_plus_delta])
 
         newbysample = tensorlib.product(allfac, axis=0)
+        if self.clip_sample_data is not None:
```

Comment:
@kratsg this is probably something that should go into a new PR rather than this one to avoid delaying this, but if we allow for this should we add logging that explicitly warns the user about what they are doing? We're mirroring ROOT's behavior here, but potentially allowing for users who might be a bit naive to wander into unexpected territory. Thoughts from @lukasheinrich and @alexander-held here?

Reply:
A warning sounds like a good idea to me. I think there's a chance that using this becomes the standard and people copy&paste things around without paying much attention to this kwarg. Having a warning could help ensure that users are aware of what is happening. What would perhaps be most useful is a message that is only emitted when the clipping does something, but that presumably requires an additional runtime check that might be expensive.

Reply:
A problem with a warning is that this function gets called a lot of times... so we don't want to spam the user. We could add a warning during model initialization...

Reply:
Added warning to initialization of model.

Reply:
Thanks! (just saw that you did this)